diagnostics/diag.go

```diff
@@ -98,7 +98,13 @@ func (di *DiagInfo) Marshal() []byte {
 }
 
 func (d *Diagnostics) getPeers() []peer.ID {
-	return d.host.Network().Peers()
```

Comment:
Since the output of `Peers()` doesn't provide the contract you want, why not write the IDs to a map so that `getPeers()` _does_?